### PR TITLE
Avoid recomputing `amountIn` when `exactIn` and `amountRemainingLessF…

### PR DESCRIPTION
**Pull request by shuhuiluo**
…ee < amountIn`

Originally part of #280.

## Related Issue
Which issue does this pull request resolve?

## Description of changes
Simplify the flow in SwapMath library by removing unnecessary calculations related to input amounts. The removed calculations are replaced with a direct assignment of the remaining amount which simplifies the logic. Moreover, the potential for any rounding issues is significantly reduced.

In the case of `exactIn` and `amountRemainingLessFee < amountIn`, the original logic computes `sqrtPriceNextX96` from `amountRemainingLessFee`. It then uses `sqrtPriceNextX96` to compute `amountIn`, which shouldn't differs much from `amountRemainingLessFee`, while `amountIn` can simply exhaust `amountRemainingLessFee`.

### CODE REVIEW

**Review by hensha256 — DISMISSED** [review]

**Comment by hensha256** [comment]
@snreynolds is also going to review this. Please dont merge without approvals from both of us 🙏

**Review by snreynolds — DISMISSED** [review]
makes sense to me! that edge case I believe is really just a rounding error and with this change it rounds the fee down which I believe is more accurate behavior?

**Review by hensha256 — APPROVED** [review]

**Review by snreynolds — APPROVED** [review]

**Pull request merged**
